packages/eslint-plugin/docs/rules/strict-boolean-expressions.md

```diff
@@ -124,6 +124,12 @@ Allows `number | null | undefined` in a boolean context.
 This is unsafe because nullable numbers can be either a falsy number or nullish.
 Set this to `true` if you don't mind implicitly treating zero or NaN the same as a nullish value.
 
+### `allowNullableEnum`
+
+Allows `number | null | undefined` in a boolean context.
```

Comment:
> `number`

It could be confusing for someone reading this how the code snippet is the same as `allowNullableNumber`. How about...

```suggestion
Allows `enum | null | undefined` in a boolean context.
```